js/bundle.go

```diff
@@ -39,6 +40,23 @@ type Bundle struct {
 	moduleResolver  *moduleResolver
 }
 
+// TODO: this is to be removed once this is not a warning and it can be moved to the registry
+func (b *Bundle) checkMetricNamesForPrometheusCompatibility() {
+	const (
+		nameRegexString = "^[a-zA-Z_][a-zA-Z0-9_]{1,128}$"
+		badNameWarning  = "Metric name should only include ASCII letters, numbers and underscores. " +
+			"This name will stop working in k6 v0.49.0."
```

Comment:
```suggestion
			"This name will stop working in k6 v0.48.0 (around December 2023)."
```
It should clarify expectations. TBH, I think it is still too relaxed.